src/runtime/converter.cs

```diff
@@ -800,32 +808,52 @@ private static bool ToArray(IntPtr value, Type obType, out object result, bool s
             Type elementType = obType.GetElementType();
             result = null;
 
-            bool IsSeqObj = Runtime.PySequence_Check(value);
-            var len = IsSeqObj ? Runtime.PySequence_Size(value) : -1;
-
             IntPtr IterObject = Runtime.PyObject_GetIter(value);
-
-            if(IterObject==IntPtr.Zero) {
+            if (IterObject == IntPtr.Zero)
+            {
                 if (setError)
                 {
                     SetConversionError(value, obType);
                 }
+                else
+                {
+                    // PyObject_GetIter will have set an error
+                    Exceptions.Clear();
+                }
                 return false;
             }
 
             Array items;
 
             var listType = typeof(List<>);
             var constructedListType = listType.MakeGenericType(elementType);
-            IList list = IsSeqObj ? (IList) Activator.CreateInstance(constructedListType, new Object[] {(int) len}) :
-                                        (IList) Activator.CreateInstance(constructedListType);
+            IList list;
+            try
+            {
+                bool IsSeqObj = Runtime.PySequence_Check(value);
+                if (IsSeqObj)
+                {
+                    var len = Runtime.PySequence_Size(value);
+                    list = (IList)Activator.CreateInstance(constructedListType, new Object[] { (int)len });
+                }
+                else
+                {
+                    list = (IList)Activator.CreateInstance(constructedListType);
+                }
+            }
+            catch
+            {
```

Comment:
Why would this fail? We should only catch specific exceptions.

Also, the exception information should be included into the resulting error unless we fully process it.

Reply:
See [Activator.CreateInstance](https://docs.microsoft.com/en-us/dotnet/api/system.activator.createinstance) for a list of the cases where it can throw.  test_generic_list_array_conversion tests one such case.  I will change the code to include the error.

Reply:
Most of them are not applicable, since the type is known to be `List<T>`

Reply:
Yes but some still are applicable.  In fact, MakeGenericType can also fail in some cases, so it should also be included in the try/catch.

Reply:
Actually, since `elementType` is the element type of an array, MakeGenericType cannot fail.  So the only possible exception seems to be ArgumentException.

Reply:
I have added the new tests test_method_with_pointer_array_argument, test_invalid_generic_type_parameter, test_invalid_generic_method_type_parameter to emphasize the importance of try/catch in these cases.